src/library/Hooks/useTimeLeft/index.tsx

```diff
@@ -117,58 +114,63 @@ export const useTimeLeft = ({
     ReturnType<typeof setInterval> | undefined
   >(undefined);
   const secIntervalRef = useRef(secInterval);
-
-  // refresh callback counter
-  let r = refreshInterval;
-
-  // refresh effects.
-  useEffect(() => {
-    // handler for handling timeleft refresh.
-    //
-    // either handle regular timeleft update, or refresh `to` via `refreshCallback` every
-    // `refreshInterval` seconds.
-    const handleRefresh = () => {
-      if (r !== 0) {
-        setTimeleft(getTimeleft());
-      } else {
-        if (refreshCallback) {
-          setStateWithRef(fromNow(refreshCallback()), setTo, toRef);
+  if (TimeleftHook) {
+    // refresh callback counter
+    let r = TimeleftHook.refreshInterval;
+
+    // refresh effects.
+    useEffect(() => {
+      // handler for handling timeleft refresh.
+      //
+      // either handle regular timeleft update, or refresh `to` via `refreshCallback` every
+      // `refreshInterval` seconds.
+      const handleRefresh = () => {
+        if (r !== 0) {
           setTimeleft(getTimeleft());
-        }
-        r = refreshInterval;
-      }
-    };
-
-    if (inLastHour()) {
-      // refresh timeleft every second.
-      if (!secIntervalRef.current) {
-        const interval = setInterval(() => {
-          if (!inLastHour()) {
-            clearInterval(secIntervalRef.current);
-            setStateWithRef(undefined, setSecInterval, secIntervalRef);
+        } else {
+          if (TimeleftHook.refreshCallback()) {
+            setStateWithRef(
+              fromNow(TimeleftHook.refreshCallback()),
+              setTo,
+              toRef
+            );
+            setTimeleft(getTimeleft());
           }
-          r = Math.max(0, r - 1);
-          handleRefresh();
-        }, 1000);
-
-        setStateWithRef(interval, setSecInterval, secIntervalRef);
-      }
-    } else {
-      setTimeleft(getTimeleft());
-      // refresh timeleft every minute.
-      if (!minIntervalRef.current) {
-        const interval = setInterval(() => {
-          if (inLastHour()) {
-            clearInterval(minIntervalRef.current);
-            setStateWithRef(undefined, setMinInterval, minIntervalRef);
-          }
-          r = Math.max(0, r - 60);
-          handleRefresh();
-        }, 60000);
-        setStateWithRef(interval, setMinInterval, minIntervalRef);
+          r = TimeleftHook.refreshInterval;
```

Comment:
This does not fall back to a default value.
`r  = timeleftHook?.refreshInterval || 60;`

can introduce defaults here too.